scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/signal/SignalApiImpl.java

```diff
@@ -121,7 +126,18 @@ public String checkForSignals(List<String> signalsSubList) throws SignalApiExcep
     public boolean sendSignal(String signalName) throws SignalApiException {
         try {
             init();
-            synchronization.compute(SIGNALS_CHANNEL, jobId, "{k, x -> x.add('" + signalName + "');x}");
+
+            // If the ready signal already exists, remove it
+            synchronization.conditionalCompute(SIGNALS_CHANNEL,
```

Comment:
if you use HashSet, you can simplify this in one line

synchronization.compute(SIGNALS_CHANNEL, jobId, x -> x.remove('" + READY_PREFIX + signalName + "'); x -> x.add('" + signalName + "');x}");